src/components/friends/async-render/iterate.ts

```diff
@@ -261,12 +273,36 @@ export function iterate(
 
 	return iter.readEls;
 
-	function setVNodeCompiler(c: AnyFunction) {
-		toVNode = c;
+	function isChildOf(vnode: VNode, id: number): boolean {
+		if (vnode.children?.[asyncRenderMarker] === id) {
+			return true;
+		}
+
+		if (Object.isArray(vnode.children)) {
+			return vnode.children.some((c) => isChildOf(Object.cast(c), id));
+		}
+
+		return false;
 	}
 
-	function setTarget(t: VNode) {
-		target = t;
+	function setVNodeCompiler(c: { wrappedCb: AnyFunction; handled?: boolean }) {
+		if (c.handled) {
+			return;
+		}
+
+		const {wrappedCb} = c;
+		toVNode = wrappedCb;
+		c.handled = true;
+		ctx.$off('[[V_FOR_CB]]', setVNodeCompiler);
+	}
+
+	function setTarget(t: { vnode: VNode }) {
+		const {vnode} = t;
+		const isChild = isChildOf(vnode, iterateId);
+		if (isChild) {
```

Comment:
Мне кажется переменная тут лишняя
```suggestion
		if (isChildOf(vnode, iterateId)) {
```